### PR TITLE
Makes association tables unique, adds organisms to geo experiments.

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Purpose/Implementation Notes

Geo experiments had no organisms.

Association tables allowed duplicate rows.

This fixes both of those issues.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

### CODE REVIEW

File: foreman/data_refinery_foreman/surveyor/array_express.py [hunk 1]
```diff
@@ -460,11 +460,6 @@ def create_samples_from_api(self,
                 original_file_sample_association.sample = sample_object
                 original_file_sample_association.save()
 
-            association = ExperimentSampleAssociation()
```
Comment (Miserlou) [inline foreman/data_refinery_foreman/surveyor/array_express.py]: Can you explain this removal? It isn't mentioned in the PR text.

File: foreman/data_refinery_foreman/surveyor/geo.py [hunk 2]
```diff
@@ -124,17 +125,17 @@ def create_experiment_and_samples_from_api(self, experiment_accession_code) -> (
 
                 all_samples.append(sample_object)
 
-                association = ExperimentSampleAssociation()
-                association.experiment = experiment_object
-                association.sample = sample_object
-                association.save()
+                ExperimentSampleAssociation.objects.get_or_create(
```
Comment (Miserlou) [inline foreman/data_refinery_foreman/surveyor/geo.py]: If you want use this pattern, can you make it universal in the codebase?

File: foreman/data_refinery_foreman/surveyor/array_express.py (hunk 1 above)
Comment (kurtwheeler) [reply foreman/data_refinery_foreman/surveyor/array_express.py]: It was duplicated immediately below, except with in a try block.

**Comment by kurtwheeler** [comment]
@Miserlou I used that pattern everywhere it made sense. I chose not to use it in places where the control flow of the `try/except` blocks were useful.

**Pull request merged**